tests/pin.go

```diff
@@ -506,6 +542,47 @@ func assertPinLsAllConsistency(t *testing.T, ctx context.Context, api iface.Core
 	}
 }
 
+func assertIsPinned(t *testing.T, ctx context.Context, api iface.CoreAPI, p path.Path, typeStr string) {
+	t.Helper()
+	withType, err := opt.Pin.IsPinned.Type(typeStr)
+	if err != nil {
+		panic("unhandled pin type")
```

Comment:
I am not sure it is elegant to panic on a test like this. It very much looks like a normal test error condition.

Reply:
I used a `panic()` because all the originating call have the type hard-coded, meaning that an error here would be a developer error (ie typo) rather than a real test failure. But that's a big gray area, I'm changing it.